name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/earnings/EarningsViewModel.java

```diff
@@ -245,6 +260,77 @@ private void calculate()
 
         Client filteredClient = clientFilter.getSelectedFilter().filter(client);
 
+        EnumSet<Mode> processGainTx = EnumSet.of(Mode.GAIN_REALIZED, Mode.GAIN_UNREALIZED, Mode.ALL);
+        if (processGainTx.contains(mode))
+        {
+            List<Trade> trades = null;
+            ClientSnapshot clientSnapshot = null;
+
+            switch (mode)
+            {
+                case GAIN_REALIZED:
+                    trades = collectTrade(filteredClient);
+                    break;
+                case GAIN_UNREALIZED:
+                    clientSnapshot = ClientSnapshot.create(filteredClient, converter, LocalDate.now());
+                    break;
+                case ALL:
+                    trades = collectTrade(filteredClient);
+                    clientSnapshot = ClientSnapshot.create(filteredClient, converter, LocalDate.now());
+                    break;
+                default:
+
+            }
+
+            if (trades != null)
+            {
+                for (Trade trade : trades)
+                {
+                    if (!trade.getEnd().isPresent())
+                        continue;
+                    if (!checkIsInIntervalTrade.test(trade))
+                        continue;
+
+                    long value = 0;
+                    value = trade.getProfitLoss().with(converter.at(trade.getEnd().get().toLocalDate())).getAmount();
+
+                    if (value != 0)
+                    {
+                        int index = (trade.getEnd().get().getYear() - startYear) * 12
+                                        + trade.getEnd().get().getMonthValue() - 1;
+                        Line line = vehicle2line.computeIfAbsent(trade.getSecurity(), s -> new Line(s, noOfmonths));
+
+                        line.values[index] += value;
+                        line.sum += value;
+
+                        sum.values[index] += value;
+                        sum.sum += value;
+                    }
+                }
+            }
+
+            if (clientSnapshot != null)
+            {
+                for (AssetPosition assetPosition : clientSnapshot.getPositionsByVehicle().values())
+                {
+                    long value = 0;
+                    value = assetPosition.getProfitLoss().getAmount();
+
+                    if (value != 0)
+                    {
+                        int index = (LocalDate.now().getYear() - startYear) * 12 + LocalDate.now().getMonthValue() - 1;
+                        Line line = vehicle2line.computeIfAbsent(assetPosition.getSecurity(),
+                                        s -> new Line(s, noOfmonths));
+                        line.values[index] += value;
```

Comment:
Der ```TradeCollector``` rechnet die Brutto-Performance und darum mit Bruttowerte (inklusive Steuern und Gebühren). Aber in den anderen Ansichten (Dividenden, Zinsen) wird mit Nettowerten gerechnet. Und insbesondere wenn man "ALL" auswählt, dann werden die Steuern und Gebühren wieder abgezogen und dann entstehen falsche Werte.

Ich denke man müsste die ```Trade#calculate``` Methode so erweitern, dass auch ```netEntryValue``` und ```netExitValue``` berechnet werden.

Reply:
Wenn ich mir das so recht überlege, ist dass vielleicht auch nicht ganz fair. Die Einnahme beim Verkauf wären dann ja grösser als tatsächlich weil die Gebühren vom Kauf fehlen. Aber den entryValue nehmen und den netExitValue, ist auch irgendwie komisch

Reply:
Vielleicht denken wir auch zu kompliziert: Bisher werden in der Ansicht "nur" die Buchungen angezeigt. Warum nicht auch "Verkaufserlöse/Kaufaufwendungen"? Dann gilt wieder "ich zahle X EUR Gebühr, habe durch den Verkauf aber auch Y EUR eingenommen". Gewinn ist was anderes (und sieht man auch in der Performance-Berechnung bzw. bei den Wertpapieren).

Reply:
> Der ```TradeCollector``` rechnet die Brutto-Performance und darum mit Bruttowerte (inklusive Steuern und Gebühren). 
Kann ich so nicht nachvollziehen. ```entryValue``` und ```exitValue``` erhalten die Werte via ```t.getTransaction().getMonetaryAmount()``` und die sind glaube ich bereits exclusive Steuern und Gebühren.
 
https://github.com/buchen/portfolio/blob/master/name.abuchen.portfolio/src/name/abuchen/portfolio/snapshot/trades/Trade.java#L45-L59

However wenn brutto und netto bei den Trades sinnvoll/notwendig/hilfreich sind, im 3. Commit @ https://github.com/buchen/portfolio/pull/1380/commits/84c64de9546bcdf30968191a9c5ac86a1e69c48d habe ich diese Möglichkeit eingebaut. Ich glaube nicht das dies hilfreich ist, aber mir fehlt hier die Erfahrung.